### ISSUE TITLE
Feature: Add volume mounts for the dind sidecar to the Runner Spec

### ISSUE THREAD

**Issue by asoldino** [summerwind/actions-runner-controller#435]
**Feature description:** Add a property on the Runner Spec to specify additional volumes that are mounted to the dind sidecar, if enabled.

**Motivation:** Having a persistent volume mounted to /var/lib/docker on the dind sidecar can improve the performance of container jobs (layer caching).

👋 would this feature make sense? 

Thanks in advance for the awesome job!

**Comment by mumoshu** [comment]
@asoldino Yeah, this would be a great addition to the controller!

Implementation-wise, we'd need to do add `dockerVolumeMounts` to RunnerSpec o as similar as other docker-sidecar-container-related fields:

```
// +optional
DockerVolumeMounts []corev1.VolumeMount `json:"dockerVolumeMounts,omitempty"`
```

https://github.com/summerwind/actions-runner-controller/blob/master/api/v1alpha1/runner_types.go#L49

Note that we can just use `volumes` for declaring additional volumes because it is per-pod configuration that can be shared among runner pod containers.

@asoldino WDYT? Would you be able to declare what you need with the above addition? Do you think we need anything more?

**Comment by asoldino** [comment]
@mumoshu This is exactly what I had in mind! What do you need from me for moving further? I think I can start working on this and create a pr myself, unless there is something else to be done first.

**Comment by mumoshu** [comment]
@asoldino I think everything's settled now and you are ready to submit a pr! (I appreciate your help 👍

**Comment by AndreaSoldino-TomTom** [comment]
Thank you @mumoshu for your input! Created pr https://github.com/summerwind/actions-runner-controller/pull/439

**Comment by asoldino** [comment]
Thank you @mumoshu for your input! Created pr #439

**Comment by peimanja** [comment]
Not sure if it's related to this issue or not but I am trying to use [aevea/action-kaniko](https://github.com/marketplace/actions/kaniko-builder) while using a runner with below template:

```yml
  spec:
    replicas: 2
    template:
      spec:
        image: summerwind/actions-runner-dind
        dockerdWithinRunnerContainer: true
        organization: myorg
        resources:
          requests:
            memory: 512Mi
            cpu: 500m
          limits:
            memory: 2Gi
            cpu: 1
        volumeMounts:
        - mountPath: /runner
          name: runner
        volumes:
        - name: runner
          emptyDir: {}
```

and I am seeing this error:
```
 Status: Downloaded newer image for gcr.io/kaniko-project/executor:debug
   ---> 7def25a8de00
  Step 4/9 : SHELL ["/busybox/sh", "-c"]
  failed to copy xattrs: failed to set xattr "security.selinux" on /var/lib/docker/volumes/b9a4672c5732a8bdeb5487f04afe6de9c93db0100dc01ca3558697d77252feb6/_data/[: permission denied
  Warning: Docker build failed with exit code 1, back off 4.543 seconds before retry.
  /usr/local/bin/docker build -t 60e226:b98c28b984844d949c888919198313c5 -f "/runner/_work/_actions/aevea/action-kaniko/master/Dockerfile" "/runner/_work/_actions/aevea/action-kaniko/master"
 ```

**Comment by mumoshu** [comment]
@peimanja To the best of my knowledge, I have no concrete idea how yours could relate to this issue(additional volume/mounts to docker container).

AFAIK any docker-backed GitHub Action is docker-build on demand. That's why it looks like failing on `docker build -f path/to/aevea/action-kaniko/master/Dockerfile` 

Also, the error seem to be a docker bug that had been fixed in https://github.com/moby/moby/pull/38316 (https://github.com/moby/moby/commit/f34727f704852ab8c51d674881f9be5da52339b4) which should already be contained in docker 19.03, which is included in `summerwind/actions-runner-dind`.

I've also found https://github.com/genuinetools/img/issues/45#issuecomment-457586746, which says it may happen with older builtkit with xfs. Is your host and hence your emptyDir is backed by xfs, by any chance?
